### PR TITLE
Download srs file first

### PR DESCRIPTION
**Pull request by ImJeremyHe**
### This PR:
Download the srs file when building the docker image.

In this [repo](https://github.com/alxiong/ark-srs), srs files are downloaded to a random temporary path first.

https://github.com/alxiong/ark-srs/blob/8ae41e79b7a209abc3855179e7c6fc885bb4680a/src/load.rs#L57C1-L75C2

Thus, it is impossible for us to add it to the `allowed_files` in sgx manifest.

### How to test this PR
With this sgx manifest
```
sys.enable_sigterm_injection = true
sys.enable_extra_runtime_domain_names_conf = true

sgx.nonpie_binary = true
sgx.enclave_size = "16G"
sgx.edmm_enable = true
sgx.remote_attestation = "dcap"
sys.experimental__enable_flock = true
sgx.use_exinfo = true

############################# SGX: ALLOWED FILES ###############################
sgx.allowed_files = ["file:/home/user/.arbitrum", "file:/home/user/kzg10-aztec20-srs-1024.bin"]
```

to see if the batch poster can work in SGX environment.

### CODE REVIEW

File: Dockerfile.sgx-poster [hunk 1]
```diff
@@ -2,8 +2,10 @@ FROM ghcr.io/espressosystems/nitro-espresso-integration/nitro-node-dev:20241124-
 
 COPY ./config /config/
 COPY ./l1keystore /l1keystore/
+RUN curl -L -o kzg10-aztec20-srs-1024.bin https://github.com/EspressoSystems/ark-srs/releases/download/v0.2.0/kzg10-aztec20-srs-1024.bin
```
Comment (sveitser) [inline Dockerfile.sgx-poster]: I think this SRS file is only for testing. Sequencer is using degree 1048584

https://github.com/EspressoSystems/espresso-sequencer/blob/dbbe97e6984f3205a5d4628b392c7f55f56a20a6/docker/sequencer.Dockerfile#L9

Comment (Sneh1999) [reply Dockerfile.sgx-poster]: Ahh thats a good catch @sveitser!

Comment (ImJeremyHe) [reply Dockerfile.sgx-poster]: Ah is this branch for production? I thought `sgx-test` in `nitro-testnode` must use the test srs file

Comment (Sneh1999) [reply Dockerfile.sgx-poster]: No this Dockerfile, should be for prod. I know we should probably merge this dockerfile with celestia-integration in nitro-espresso-integration repo. I havent gotten the time to do that till now.

Comment (Sneh1999) [reply Dockerfile.sgx-poster]: For now, I think its okay we merge this here for testing and we can merge the final version in nitro-espresso-integration once the testing is complete

**Pull request merged**